### ISSUE TITLE
Github Readme Analyzer: Failed:('max_depth') error

### ISSUE THREAD

**Issue by wbobeirne** [algorithmiaio/awesome-ipython-notebooks#1]
Just found your github analyzer tool, but was disappointed to see every repository I tried resulted in a `Failed:('max_depth')` error. I know this is a year old, but thought I'd drop a note anyway.

**Comment by besirkurtulmus** [comment]
@wbobeirne thanks for letting us know.

Github might have changed their API.

I'll push a fix once I determine the problem.

**Comment by besirkurtulmus** [comment]
@wbobeirne: I was unable to reproduce the error. I'm getting recommendations for any Github repo I provide.

Can you please provide a stack trace, and also what repo's you've used?

**Comment by jgauffin** [comment]
here is my request/response (HAR-file in google chrome):

```javascript
{
  "log": {
    "version": "1.2",
    "creator": {
      "name": "WebInspector",
      "version": "537.36"
    },
    "pages": [],
    "entries": [
      {
        "startedDateTime": "2017-08-22T03:46:31.781Z",
        "time": 1438.4620000055293,
        "request": {
          "method": "POST",
          "url": "https://api.algorithmia.com/v1/web/algo/nlp/AnalyzeGithubReadme",
          "httpVersion": "HTTP/1.1",
          "headers": [
            {
              "name": "Authorization",
              "value": "Simple ***REMOVED BY JGAUFFIN***"
            },
            {
              "name": "Origin",
              "value": "http://demos.algorithmia.com"
            },
            {
              "name": "Accept-Encoding",
              "value": "gzip, deflate, br"
            },
            {
              "name": "Host",
              "value": "api.algorithmia.com"
            },
            {
              "name": "Accept-Language",
              "value": "en-US,en;q=0.8,sv;q=0.6"
            },
            {
              "name": "User-Agent",
              "value": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.101 Safari/537.36"
            },
            {
              "name": "Content-Type",
              "value": "application/json"
            },
            {
              "name": "Accept",
              "value": "application/json, text/javascript"
            },
            {
              "name": "Referer",
              "value": "http://demos.algorithmia.com/github-readme-analyzer/"
            },
            {
              "name": "Connection",
              "value": "keep-alive"
            },
            {
              "name": "Content-Length",
              "value": "22"
            }
          ],
          "queryString": [],
          "cookies": [],
          "headersSize": 560,
          "bodySize": 22,
          "postData": {
            "mimeType": "application/json",
            "text": "\"jgauffin/markdownweb\""
          }
        },
        "response": {
          "status": 200,
          "statusText": "OK",
          "httpVersion": "HTTP/1.1",
          "headers": [
            {
              "name": "Date",
              "value": "Tue, 22 Aug 2017 03:46:34 GMT"
            },
            {
              "name": "Content-Encoding",
              "value": "gzip"
            },
            {
              "name": "Vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "X-Frame-Options",
              "value": "DENY"
            },
            {
              "name": "Strict-Transport-Security",
              "value": "max-age=0; includeSubDomains; preload"
            },
            {
              "name": "Content-Type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "Access-Control-Allow-Origin",
              "value": "*"
            },
            {
              "name": "Connection",
              "value": "keep-alive"
            },
            {
              "name": "Content-Length",
              "value": "88"
            }
          ],
          "cookies": [],
          "content": {
            "size": 71,
            "mimeType": "application/json",
            "compression": -17,
            "text": "{\"error\":{\"message\":\"'max_depth'\"},\"metadata\":{\"duration\":1.288624783}}"
          },
          "redirectURL": "",
          "headersSize": 315,
          "bodySize": 88,
          "_transferSize": 403
        },
        "cache": {},
        "timings": {
          "blocked": 0.369000001228415,
          "dns": -1,
          "connect": -1,
          "send": 0.09099999442696505,
          "wait": 1437.1200000023246,
          "receive": 0.8820000075493226,
          "ssl": -1
        },
        "serverIPAddress": "52.22.137.59",
        "connection": "306515"
      }
    ]
  }
}
```

**Comment by wbobeirne** [comment]
No stack trace, I was running it directly off of http://demos.algorithmia.com/github-readme-analyzer/. It's failing on all of my repos, but `wbobeirne/stranger-things` for instance.

**Comment by besirkurtulmus** [comment]
It seems that I've misunderstood the initial question.

I went through the whole iPython notebook and couldn't find an issue. I thought @wbobeirne was having an issue with the source-code since this is an issue page for the project.

I've fixed the issue, the demo site should be working now.

For reference, the issue was a scikit-learn version compatibility problem. The container running the demo was using a newer version compared to the trained version.

I pegged the version number to 0.17.1, which should be the latest stable version when the demo first came out.
